kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/util/LinuxNetworkUtil.java

```diff
@@ -264,22 +269,66 @@ private boolean isWifiLinkUpInternal(String ifaceName) throws KuraException {
     }
 
     public static boolean toolExists(String tool) {
-        boolean ret = false;
-        final String[] searchFolders = new String[] { "/sbin/", "/usr/sbin/", "/bin/", "/usr/bin/" };
+        return getToolPath(tool).isPresent();
+    }
 
-        if (TOOLS.contains(tool)) {
-            ret = true;
+    public static Optional<String> getToolPath(String tool) {
+        Optional<String> optionalToolPath = getCachedTool(tool);
+        if (optionalToolPath.isPresent()) {
+            return optionalToolPath;
         } else {
-            for (String folder : searchFolders) {
-                File fTool = new File(folder + tool);
-                if (fTool.exists()) {
-                    TOOLS.add(tool);
-                    ret = true;
-                    break;
+            for (String folder : DEFAULT_PATH) {
+                String toolPath = folder + tool;
+                File toolFile = new File(toolPath);
+                if (toolFile.exists()) {
+                    TOOLS.add(toolPath);
+                    return Optional.of(toolPath);
                 }
             }
         }
-        return ret;
+        return Optional.empty();
+    }
+
+    private static Optional<String> getCachedTool(String tool) {
+        return TOOLS.stream().filter(item -> {
+            return item.endsWith(tool);
+        }).findFirst();
+    }
+
+    /**
+     * Checks if the given Systemd system unit is installed.
+     * The result is based on the exit code of "systemctl status <unitName>"
+     * as presented in https://www.man7.org/linux/man-pages/man1/systemctl.1.html#EXIT_STATUS
+     *
+     * @param unitName
+     *            the name of the Systemd system unit
+     * @return true if the unit is installed
+     */
+    public static boolean systemdSystemUnitExists(String unitName) {
+        Optional<String> optionalSystemctlPath = getToolPath("systemctl");
+        if (!optionalSystemctlPath.isPresent()) {
+            logger.debug("Systemctl command not found in default paths");
+            return false;
+        }
+        PROCESS_BUILDER.command(optionalSystemctlPath.get(), "status", unitName);
+        Process process;
+        try {
+            process = PROCESS_BUILDER.start();
+            StreamGobbler streamGobbler = new StreamGobbler(process.getInputStream(), logger::debug);
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            Future<?> future = executor.submit(streamGobbler);
+
+            int exitCode = process.waitFor();
+            future.get(10, TimeUnit.SECONDS);
+            return exitCode < 4;
+        } catch (IOException | ExecutionException | TimeoutException e) {
+            logger.error(String.format("Cannot check %s unit existence", unitName), e);
```

Comment:
You should avoid using `String.format` in the logger parameters, it is inefficient since it is computed even if error logging is disabled (cannot think of such a case in real world).

You can safely write:

```java
logger.error("Cannot check {} unit existence", unitName, e);
```

As of log4j 1.6 (we are on 1.7+), we can parametrize the arguments in the presence of an exception, assuming the exception is the last parameter

https://www.slf4j.org/faq.html#paramException
https://www.slf4j.org/apidocs/org/slf4j/Logger.html